### PR TITLE
Issue #4502: Update documentation: replaced deprecated NEIGHBORS() function

### PR DESCRIPTION
**Pull request by goedderz**

### CODE REVIEW

**Comment by Simran-B** [comment]
To make the replacement look less complicated, I guess we could remove some of the line breaks and perhaps replace the query options with a DISTINCT?

```
...> "    FOR author, writtenBy IN INBOUND b._id written " +
...> "    RETURN DISTINCT { " +
...> "        vertex: author, " +
...> "        edge: writtenBy " +
...> "    } " +
```

**Comment by dothebart** [comment]
nb, if you only change documentation, you should name the branch `documentation/xxx` or `documentation-3.3/xxx`

**Comment by goedderz** [comment]
@dothebart thanks, will do.

@Simran-B I simplified the queries as you suggested. I also swapped the first and second example in order to have the simplest one first.

**Comment by goedderz** [comment]
#4502

File: Documentation/Books/Cookbook/AQL/Joins.md [hunk 1]
```diff
@@ -359,194 +359,209 @@ arangosh> db.written.save("authors/2938210813",
...
-            "first" : "Maxima", 
-            "last" : "Musterfrau" 
-          } 
-        } 
-      } 
-    ] 
-  } 
+...> "FOR b IN books " +
+...> "LET authorsByBook = ( " +
+...> "    FOR author IN INBOUND b._id written" +
```
Comment (mchacki) [inline Documentation/Books/Cookbook/AQL/Joins.md]: Please use:
`FOR author IN INBOUND b written OPTIONS {bfs: true, uniqueVertices: "global"} RETURN author`

As a query here. That is much more performant than the distinct variant. as the later cannot be optimized (yet)

File: Documentation/Books/Cookbook/AQL/Joins.md [hunk 2]
```diff
@@ -359,194 +359,209 @@ arangosh> db.written.save("authors/2938210813",
...
-          "first" : "Maxima", 
-          "last" : "Musterfrau" 
-        } 
-      } 
-    ] 
-  } 
+...> "FOR b IN books " +
+...> "LET authorsByBook = ( " +
+...> "    FOR author, writtenBy IN INBOUND b._id written " +
+...> "    RETURN DISTINCT { " +
```
Comment (mchacki) [inline Documentation/Books/Cookbook/AQL/Joins.md]: No need for DISTINCT here. Each edge of an author will be visited only once.

`b._id` could be replaced by `b`.

File: Documentation/Books/Cookbook/AQL/Joins.md [hunk 3]
```diff
@@ -359,194 +359,209 @@ arangosh> db.written.save("authors/2938210813",
...
-          "first" : "Maxima", 
-          "last" : "Musterfrau" 
-        } 
-      } 
-    ] 
-  } 
+...> "FOR b IN books " +
+...> "LET authors = ( " +
+...> "    FOR author IN INBOUND b._id written " +
+...> "    RETURN DISTINCT author " +
```
Comment (mchacki) [inline Documentation/Books/Cookbook/AQL/Joins.md]: See `OPTIONS {bfs: true, uniqueVertices: "global"}` as above.

File: Documentation/Books/Cookbook/AQL/Joins.md [hunk 4]
```diff
@@ -359,194 +359,209 @@ arangosh> db.written.save("authors/2938210813",
...
-        "_key" : "2980088317", 
-        "title" : "The beauty of JOINS" 
-      } 
-    ] 
-  } 
+> db._query(
+...> "FOR a IN authors " +
+...> "LET booksByAuthor = ( " +
+...> "    FOR b IN OUTBOUND a._id written " +
+...> "    RETURN DISTINCT b" +
```
Comment (mchacki) [inline Documentation/Books/Cookbook/AQL/Joins.md]: OPTIONS see above

**Comment by goedderz** [comment]
@mchacki The queries are changed as you suggested.

**Pull request merged**